src/track/globaltrackcache.cpp

```diff
@@ -74,7 +79,21 @@ void deleteTrack(Track* plainPtr) {
                 << "Deleting"
                 << plainPtr;
     }
+    DEBUG_ASSERT(plainPtr->signalsBlocked());
+#if QT_VERSION >= QT_VERSION_CHECK(5, 6, 0)
+    if (plainPtr->thread()->loopLevel() > 0) {
+        plainPtr->deleteLater();
```

Comment:
https://doc.qt.io/qt-5/qthread.html:
> Note: This can only be called within the thread itself, i.e. when it is the current thread.

Is this guaranteed?

Reply:
Yes. Cache entries are always evicted on the main (event loop) thread, and then the track pointer goes out of scope.

Reply:
I found a correct, more explicit solution. The tests now provide a function pointer to override the default `QObject::deleteLater()` deletion.

Reply:
This new solution is independent of the Qt version.